### PR TITLE
made LEDs for the shoot command🦟™️

### PR DESCRIPTION
**Pull request by BrodyKarr**

### CODE REVIEW

File: src/main/java/frc/robot/Constants.java [hunk 1]
```diff
@@ -6,7 +6,21 @@
 
 /** Add your docs here. */
 public class Constants {
+<<<<<<< Updated upstream
```
Comment (TaylerUva) [inline src/main/java/frc/robot/Constants.java]: Delete

File: src/main/java/frc/robot/Constants.java [hunk 2]
```diff
@@ -6,7 +6,21 @@
...
     public static class constDrivetrain {
       public static final double SLOW_MODE_MULTIPLIER = 0.5;
     }
+=======
+  public static class constDrivetrain {
+
+  }
+
```
Comment (TaylerUva) [inline src/main/java/frc/robot/Constants.java:17]: Delete

File: src/main/java/frc/robot/Constants.java [hunk 3]
```diff
@@ -6,7 +6,21 @@
...
+  public static class constLED {
+    public static final double LED_BRIGHTNESS = 1;
+    public static final int LED_NUMBER = 192;
+    public static final int LED_STRIP_START_INDEX = 8;
+    public static final int[] CLEAR_LEDS = { 0, 0, 0 };
+    public static final int[] LED_Shooting = { 105, 69, 9 };
+  }
+>>>>>>> Stashed changes
```
Comment (TaylerUva) [inline src/main/java/frc/robot/Constants.java]: Delete

**Review by TaylerUva — CHANGES_REQUESTED** [review]

File: src/main/java/frc/robot/RobotContainer.java [hunk 4]
```diff
@@ -43,11 +48,15 @@ public RobotContainer() {
...
-    m_driverController.btn_RightTrigger.onTrue(com_Shoot);
+    m_driverController.btn_B.whileTrue(com_IntakeGround);
+    m_driverController.btn_LeftBumper.whileTrue(new intakeHopper(subHopper, subStager, subLED));
+    m_driverController.btn_X.whileTrue(com_PrepShooter);
+    m_driverController.btn_A.whileTrue(com_StageGP);
+    m_driverController.btn_LeftTrigger.whileTrue(com_IntakeHopper);
+    m_driverController.btn_RightBumper.whileTrue(new EjectGP(subIntake, subHopper, subLED));
+
+    m_driverController.btn_Y.onTrue(com_Shoot);
+    m_driverController.btn_RightBumper.whileTrue(com_EjectShooter);
```
Comment (TaylerUva) [inline src/main/java/frc/robot/RobotContainer.java]: You altered the button mappings from what they should be

File: src/main/java/frc/robot/commands/Drive.java [hunk 5]
```diff
@@ -5,42 +5,48 @@
...
+  public void initialize() {
+  }
 
   // Called every time the scheduler runs while the command is scheduled.
   @Override
   public void execute() {
-    globalDrivetrain.setDrivetrainSpeed(globalForwardSpeed.getAsDouble(), globalRotationSpeed.getAsDouble(), globalSlowMode.getAsBoolean());
+    globalDrivetrain.setDrivetrainSpeed(globalForwardSpeed.getAsDouble(), globalRotationSpeed.getAsDouble(),
+        globalSlowMode.getAsBoolean());
+    globalLED.setLEDs(constLED.LED_DRIVE);
```
Comment (TaylerUva) [inline src/main/java/frc/robot/commands/Drive.java]: Drive is always running so it should not have any LEDs

File: src/main/java/frc/robot/commands/Shoot.java [hunk 6]
```diff
@@ -6,29 +6,37 @@
...
+    if ((globalShooter.getPropelMotorVelocity() >= 0.6) && (globalShooter.getSpiralMotorVelocity() >= 0.8)) {
+      globalStager.setStagerMotorVelocity(0.3);
+      globalLED.setLEDs(constLED.LED_SHOOTING);
+      if ((globalShooter.getPropelMotorVelocity() >= Constants.constShooter.PROPEL_MOTOR_VELOCITY)
+          && (globalShooter.getSpiralMotorVelocity() >= Constants.constShooter.SPIRAL_MOTOR_VELOCITY)) {
+        globalStager.setStagerMotorVelocity(Constants.constStager.STAGER_MOTOR_VELOCITY);
+        globalStager.setTopStagerMotorVelocity(Constants.constStager.TOP_STAGER_MOTOR_VELOCITY);
+      } else {
+        globalStager.setStagerMotorVelocityNuetralOutput();
+      }
```
Comment (TaylerUva) [inline src/main/java/frc/robot/commands/Shoot.java]: You undid the constants with the merge conflict resolution

**Review by TaylerUva — CHANGES_REQUESTED** [review]
Overall very good just a few issues

File: src/main/java/frc/robot/commands/intakeHopper.java [hunk 7]
```diff
@@ -6,25 +6,31 @@
...
     this.subStager = subStager;
-
+    subLED = intakeHopperLED;
   }
 
   // Called when the command is initially scheduled.
   @Override
   public void initialize() {
+
+    subHopper.setOrientationMotorSpeed(.5);
```
Comment (BrodyKarr) [inline src/main/java/frc/robot/commands/intakeHopper.java]: delete line 32

File: src/main/java/frc/robot/commands/EjectGP.java [hunk 8]
```diff
@@ -5,23 +5,30 @@
...
+    this.globalHopper = subHopper;
     this.globalIntake = subIntake;
+    globalLED = ejectLED;
   }
 
   // Called when the command is initially scheduled.
   @Override
   public void initialize() {
-    globalIntake.setIntakeVelocity(Constants.constIntake.INTAKE_EJECT_VELOCITY);
+    globalIntake.setIntakeVelocity(-0.5);
```
Comment (BrodyKarr) [inline src/main/java/frc/robot/commands/EjectGP.java:30]: replace this line with globalIntake.setIntakeVelocity(Constants.constIntake.INTAKE_EJECT_VELOCITY)

**Review by TaylerUva — APPROVED** [review]

**Pull request merged**